### ISSUE TITLE
test_gdb: test_pycfunction_fastcall_keywords() failed on PPC64LE Fedora Stable Clang 3.x

### ISSUE THREAD

**Issue by vstinner** [python/cpython#110166]
PPC64LE Fedora Stable Clang 3.x:

```
======================================================================
FAIL: test_pycfunction_fastcall_keywords (test.test_gdb.test_cfunction_full.CFunctionFullTests.test_pycfunction_fastcall_keywords) [_testcapi.meth_fastcall_keywords]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/buildbot/buildarea/3.x.cstratak-fedora-stable-ppc64le.clang/build/Lib/test/test_gdb/test_cfunction.py", line 67, in check_pycfunction
    self.check(func_name, cmd)
  File "/home/buildbot/buildarea/3.x.cstratak-fedora-stable-ppc64le.clang/build/Lib/test/test_gdb/test_cfunction_full.py", line 32, in check
    self.assertRegex(gdb_output, regex)
AssertionError: Regex didn't match: '#(1|2)\\ <built\\-in\\ method\\ meth_fastcall_keywords' not found in 'Breakpoint 1 (meth_fastcall_keywords) pending.\n
[Thread debugging using libthread_db enabled]\n
Using host libthread_db library "/lib64/libthread_db.so.1".\n
\n
Breakpoint 1, meth_fastcall_keywords (self=<module at remote 0x7fffea3948f0>, args=, nargs=0, kwargs=0x0) at ./Modules/_testcapimodule.c:2114\n
2114\t    PyObject *pyargs = _fastcall_to_tuple(args, nargs);\n
Unable to locate python frame\n
'

Stdout:

  test call: _testcapi.meth_fastcall_keywords()

======================================================================
FAIL: test_pycfunction_varargs (test.test_gdb.test_cfunction_full.CFunctionFullTests.test_pycfunction_varargs) [_testcapi.meth_varargs]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/buildbot/buildarea/3.x.cstratak-fedora-stable-ppc64le.clang/build/Lib/test/test_gdb/test_cfunction.py", line 67, in check_pycfunction
    self.check(func_name, cmd)
  File "/home/buildbot/buildarea/3.x.cstratak-fedora-stable-ppc64le.clang/build/Lib/test/test_gdb/test_cfunction_full.py", line 32, in check
    self.assertRegex(gdb_output, regex)
AssertionError: Regex didn't match: '#(1|2)\\ <built\\-in\\ method\\ meth_varargs' not found in 'Breakpoint 1 (meth_varargs) pending.\n
[Thread debugging using libthread_db enabled]\n
Using host libthread_db library "/lib64/libthread_db.so.1".\n
\n
Breakpoint 1, meth_varargs (self=<module at remote 0x7fffea394890>, args=()) at ./Modules/_testcapimodule.c:2067\n
2067\t    return Py_BuildValue("NO", _null_to_none(self), args);\n
Unable to locate python frame\n
'

Stdout:

  test call: _testcapi.meth_varargs()

----------------------------------------------------------------------
```

configure:

```
./configure --prefix '$(PWD)/target' --with-pydebug
```

test.pythoninfo:

```
CC.version: gcc (GCC) 13.2.1 20230728 (Red Hat 13.2.1-1)

libregrtests.build_info: debug

sysconfig[PY_CFLAGS]: -fno-strict-overflow -Wsign-compare -g -Og -Wall
sysconfig[PY_CFLAGS_NODIST]: -std=c11 -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-function-declaration -fvisibility=hidden -I./Include/internal
sysconfig[PY_STDMODULE_CFLAGS]: -fno-strict-overflow -Wsign-compare -g -Og -Wall -std=c11 -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-function-declaration -fvisibility=hidden -I./Include/internal -I. -I./Include
```

build: https://buildbot.python.org/all/#/builders/435/builds/3683

### Linked PRs
* gh-110331

**Comment by vstinner** [comment]
That's a variant of gh-104736.

code.py:

```py
import _testcapi      
def foo(): 
    _testcapi.meth_fastcall_keywords()
def bar():                    
    foo()                                
bar()                  
```

Gdb:

```
[vstinner@localhost cpython]$ gdb -args ./python code.py
GNU gdb (GDB) Fedora Linux 13.2-5.fc38
(gdb) b meth_fastcall_keywords
(gdb) source python-gdb.py
(gdb) run
Starting program: /home/vstinner/cpython/python code.py

Breakpoint 1, meth_fastcall_keywords (self=<module at remote 0x7fffea37c590>, args=0x7fffea410128, nargs=0, kwargs=0x0) at ./Modules/_testcapimodule.c:2114
2114	    PyObject *pyargs = _fastcall_to_tuple(args, nargs);

(gdb) py-bt-full
Unable to locate python frame

(gdb) where
#0  meth_fastcall_keywords (self=<module at remote 0x7fffea37c590>, args=0x7fffea410128, nargs=0, kwargs=0x0) at ./Modules/_testcapimodule.c:2114
#1  0x0000000000000000 in ?? ()
```

**Comment by vstinner** [comment]
Using ``clang -O0``, it works as expected:

```
./configure CC=clang LD=clang --with-pydebug 'CFLAGS=-fno-omit-frame-pointer -O0'
(...)
All 5 tests OK.

Total duration: 1 min 28 sec
Total tests: run=57 skipped=1
Total test files: run=5/5
Result: SUCCESS
```

**Comment by vstinner** [comment]
The issue should now be fixed in 3.11, 3.12 and main branches.
